unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,225 @@
+            f"{AREA_ID}/unregister_participant",
+            json.dumps(data))
+
+    def test_unregister_is_active_false_throws_exception(self,
+                                                         redis_client_auto_register):
+        """Checks if is active set to false throws exception when user is already unregistered."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already unregistered from the market."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = False
+            self.redis_client.unregister(is_blocking=True)
+
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(self,
+                                                        mock_wait_until_timeout_blocking,
+                                                        redis_client_auto_register):
+        """Checks if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = True
+            self.redis_client.unregister(is_blocking=True)
+
+    def test_on_register(self, redis_client_auto_register):
+        """Assigning the correct arguments to blocking command response and
+           checking by calling on_register function with correct message
+           and should not throw an exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "register": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_register(message)
+
+    def test_on_unregister(self, redis_client_auto_register):
+        """Assigning the correct arguments to blocking command response and
+           checking by calling on_unregister function with correct message
+           and should not throw an exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID,
+                "response": "success"}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "unregister": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_unregister(message)
+
+    def test_on_unregister_throws_exception(self, redis_client_auto_register):
+        """Checks if response is not success should throw exception """
```

Comment:
```suggestion
        """Check if exception is raised when response of on_unregister is not successful."""
```

Reply:
Sure, I will adapt to these changes. I will modify to other docstrings as well. Thank you.